src/cliConfigChange.ts

```
@@ -1,22 +1,75 @@
+};
+export const fileCopy = function(srcFilePath:string, dstFilePath:string){
+    try{
+        //incase sometimes the config file has the red only we are changing the file permissions.
+        fs.chmodSync(dstFilePath, 0o755); 
+        fs.copyFileSync(srcFilePath, dstFilePath);
+    }catch(err:any){
+        throw Error(`Failed to copy the contents of file from : ${srcFilePath} to : ${dstFilePath} due to - ${err.toString()}`);
     }
-    else{
-        return false;
+};
+
+export const writeConfig = async function(){
+    try{
+        const configPath = path.resolve(os.homedir(),".akamai-cli","config");
+        const configPathOld = path.resolve(os.homedir(),".akamai-cli","config_old");
+        if(fs.existsSync(configPath)){
+            fs.copyFileSync(configPath,configPathOld);
+            const arrayStringBeforeConfigAdd = readToArrayListFromFile(configPath);
+            if(setAkamaiCLIConfig(configPath)){
+                const arrayStringAfterConfigAdd = readToArrayListFromFile(configPath);
+                //we do this becasue if some time the old array has more or vceversa this will finally give us the diff properly
+                const diff = getDifference(arrayStringAfterConfigAdd,arrayStringBeforeConfigAdd);
+                console.log("diff is" + diff);
+                if(!(diff.length === 0 || (diff.length ===1 && diff[0] === 'last-upgrade-check=ignore')))
+                {
+                    vscode.window.showErrorMessage(`Failed to Modify the akamai config file at ${configPath}. Add attribute "last-upgrade-check = ignore" to config file at ${configPath} manually.`);
+                    fs.copyFileSync(configPathOld, configPath);
+                }
+            }
+        }
+        else{
+            throw Error(`Config File path:${configPath} not found. Make sure the akamai cli config at ${configPath} is set properly and also add attribute "last-upgrade-check = ignore" at ${configPath} to keep the akamai cli working.`) ;
```

If I run `brew install akamai` and never run the `akamai` command, no config will exist.

I don't think we need to throw an error if the file isn't present.

---

since for checking the akamai --version every time when we are checking if akamai is installed or not  we use akamai command .. so technically it should exists .. if in case some user had deleted then we are displaying this error so that they will know that vscode is not responding since there is no config file or any error occurred in it .. if u still feel this is not required I will make changes accordingly.

---

OK that makes sense. Can you still remove the error throwing? If it's not present Akamai CLI should generate the file, the error seems like it exists in an edge case that shouldn't happen but would confuse the user if it did.